cmd/soroban-rpc/internal/db/event.go

```diff
@@ -78,15 +100,59 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 		}
 
 		query := sq.Insert(eventTableName).
-			Columns("id", "ledger_sequence", "application_order", "contract_id", "event_type")
+			Columns(
+				"id",
+				"contract_id",
+				"event_type",
+				"event_data",
+				"ledger_close_time",
+				"transaction_hash",
+				"topic1",
+				"topic2",
+				"topic3",
+				"topic4",
+			)
 
 		for index, e := range txEvents {
-			var contractId []byte
+
+			var contractID []byte
 			if e.Event.ContractId != nil {
-				contractId = e.Event.ContractId[:]
+				contractID = e.Event.ContractId[:]
 			}
-			id := events.Cursor{Ledger: lcm.LedgerSequence(), Tx: tx.Index, Op: 0, Event: uint32(index)}.String()
-			query = query.Values(id, lcm.LedgerSequence(), tx.Index, contractId, int(e.Event.Type))
+
+			id := Cursor{Ledger: lcm.LedgerSequence(), Tx: tx.Index, Op: 0, Event: uint32(index)}.String()
```

Comment:
why not store this value as an integer? I think it will fit within 64 bits

Reply:
You meant to store just TOID right? 

but existing [TOID](https://github.com/stellar/go/blob/master/toid/main.go#L17-L18) does not consider event index so in RPC we have Cursor which is basically TOID + event_index [this](https://github.com/stellar/soroban-rpc/blob/eb7cb8aa8ec4c4129bcb5dc0dfc8c6f963f967a5/cmd/soroban-rpc/internal/events/cursor.go#L31)

As a workaround I can replace `op_index` with `event index` as `op_index` is always 0 for events but might not be future proof?

Reply:
ok, I didn't realize that the toid + event index would not fit in 64 bits. I think it's fine to keep it as a string